sources/controllers/FriendController.ts

```diff
@@ -1,33 +1,178 @@
+
 export class FriendController {
-    public readonly inviteUser = async (_req: Request, res: Response) => {
+
+    private readonly userRepository: UserRepository;
+    private readonly userView: UserView;
+
+    constructor(userRepository: UserRepository, userView: UserView){
+        this.userRepository = userRepository;
+        this.userView = userView;
+    }
+
+    public readonly inviteUser = async (req: Request, res: Response) => {
+        const requestingUser = getRequestingUser(req);
+
+        // Validate request's parameters
+        const targetUserId = parseInt(req.params.userId, 10);
+        if (isNaN(targetUserId)) {
+            throw Boom.badRequest(`Invalid User id "${req.params.userId}"`);
+        }
+
+        if (targetUserId === requestingUser.id) {
+            throw Boom.forbidden(`You cannot invite yourself`);
+        }
+
+        // Check if user exists
+        const targetUser = await this.userRepository.getUserById(targetUserId);
+        if (!targetUser) {
+            throw Boom.notFound(`User id "${req.params.userId}" does not exist`);
+        }
+
+        // Check existing friendship in both ways
+        const existingFriendship1 = await this.userRepository.getUserFriendship(requestingUser.id, targetUser.id);
+        const existingFriendship2 = await this.userRepository.getUserFriendship(targetUser.id, requestingUser.id);
+        if (existingFriendship1 || existingFriendship2) {
+            throw Boom.forbidden("An invitation have already been sent");
+        }
+
+        // Create new invitation
+        await this.userRepository.createUserFriendship(new UserFriendship({
+            requestingId: requestingUser.id,
+            requestedId: targetUser.id,
+            confirmed: false,
+        }));
+
         res.status(200).json({
-            message: "Friend request sent",
+            message: "Invitation Sent",
         });
     }
 
-    public readonly inviteAccept = async (_req: Request, res: Response) => {
+    public readonly inviteAccept = async (req: Request, res: Response) => {
+        const requestingUser  = getRequestingUser(req);
+
+        const targetUserId = parseInt(req.params.userId, 10);
+        if (isNaN(targetUserId)) {
+            throw Boom.badRequest(`Invalid User id "${req.params.userId}"`);
+        }
+
+        // Check if user exists
+        const targetUser = await this.userRepository.getUserById(targetUserId);
+        if (!targetUser) {
+            throw Boom.notFound(`User id "${req.params.userId}" does not exist`);
+        }
+
+        if (targetUserId === requestingUser.id) {
+            throw Boom.forbidden(`You cannot accept invite from yourself`);
+        }
+
+        // Check existing friendship
+        const existingFriendship = await this.userRepository.getUserFriendship(targetUser.id, requestingUser.id);
+        if (!existingFriendship) {
+            throw Boom.forbidden("Friendship already exist");
```

Comment:
Dans ce cas tu es en train de check si une invitation a été créée (pour justement l'accepter)
```suggestion
            throw Boom.forbidden("No pending invitation from this User");
```